common/src/main/java/com/wynntils/models/players/FriendsModel.java

```diff
@@ -0,0 +1,195 @@
+    private static final Pattern FRIEND_LIST_FAIL_1 = Pattern.compile("§eWe couldn't find any friends\\.");
+    private static final Pattern FRIEND_LIST_FAIL_2 = Pattern.compile("§eTry typing §r§6/friend add Username§r§e!");
+    private static final Pattern FRIEND_REMOVE_MESSAGE_PATTERN =
+            Pattern.compile("§e(.+) has been removed from your friends!");
+    private static final Pattern FRIEND_ADD_MESSAGE_PATTERN = Pattern.compile("§e(.+) has been added to your friends!");
+
+    private static final Pattern JOIN_PATTERN = Pattern.compile(
+            "(?:§a|§r§7)(?:§o)?(.+)§r(?:§2|§8(?:§o)?) has logged into server §r(?:§a|§7(?:§o)?)(?<server>.+)§r(?:§2|§8(?:§o)?) as (?:§r§a|§r§7(?:§o)?)an? (?<class>.+)");
+    private static final Pattern LEAVE_PATTERN = Pattern.compile("(?:§a|§r§7)(.+) left the game\\.");
+
+    private boolean expectingFriendMessage = false;
+
+    private Set<String> friends;
+
+    public FriendsModel(WorldStateModel worldStateModel) {
+        super(List.of(worldStateModel));
+        resetFriendData();
+    }
+
+    @SubscribeEvent
+    public void onAuth(HadesEvent.Authenticated event) {
+        if (!Models.WorldState.onWorld()) return;
+
+        requestFriendData();
+    }
+
+    @SubscribeEvent
+    public void onWorldStateChange(WorldStateEvent event) {
+        if (event.getNewState() == WorldState.WORLD) {
+            requestFriendData();
+        } else {
+            resetFriendData();
+        }
+    }
+
+    @SubscribeEvent
+    public void onChatReceived(ChatMessageReceivedEvent event) {
+        if (event.getMessageType() != MessageType.FOREGROUND) return;
+
+        String coded = event.getOriginalCodedMessage();
+        String unformatted = ComponentUtils.stripFormatting(coded);
+
+        Matcher joinMatcher = JOIN_PATTERN.matcher(coded);
+        if (joinMatcher.matches()) {
+            WynntilsMod.postEvent(new FriendConnectionEvent.Join(joinMatcher.group(1)));
+        } else {
+            Matcher leaveMatcher = LEAVE_PATTERN.matcher(coded);
+            if (leaveMatcher.matches()) {
+                WynntilsMod.postEvent(new FriendConnectionEvent.Leave(leaveMatcher.group(1)));
+            }
+        }
+
+        if (tryParseFriendMessages(coded)) {
+            return;
+        }
+
+        if (expectingFriendMessage) {
+            if (tryParseFriendList(unformatted) || tryParseNoFriendList(coded)) {
+                event.setCanceled(true);
+                expectingFriendMessage = false;
+                return;
+            }
+
+            // Skip first message of two, but still expect more messages
+            if (FRIEND_LIST_FAIL_1.matcher(coded).matches()) {
+                event.setCanceled(true);
+                return;
+            }
+        }
+    }
+
+    private boolean tryParseNoFriendList(String coded) {
+        if (FRIEND_LIST_FAIL_2.matcher(coded).matches()) {
+            WynntilsMod.info("Player has no friends!");
```

Comment:
Ohhh... this sounds soo sad! 😿 

But seriously, maybe rephrase it to something like "Friend list is empty", so we don't trigger a lonely suicidal player to read the logs and think their life sucks.